active/swagger-secret-detector.js

```diff
@@ -0,0 +1,282 @@
+// Note that new active scripts will initially be disabled
+// -------------------------------------------------------------------
+// Swagger Secrets & Version Detector - ZAP Active Scan Rule Script
+// -------------------------------------------------------------------
+// Modern ZAP registration using getMetadata() function
+// Import required ZAP Java types for modern registration
+
+var URI = Java.type('org.apache.commons.httpclient.URI');
+var ScanRuleMetadata = Java.type("org.zaproxy.addon.commonlib.scanrules.ScanRuleMetadata");
+var CommonAlertTag = Java.type("org.zaproxy.addon.commonlib.CommonAlertTag");
+function getMetadata() {
+return ScanRuleMetadata.fromYaml(`
+id: 100001
+name: Swagger UI Secret & Vulnerability Detector
+description: >
+  Detects exposed Swagger UI and OpenAPI endpoints that leak sensitive secrets such as API keys, 
+  OAuth client secrets, access tokens, or run vulnerable versions. This scanner performs comprehensive 
+  detection of sensitive information disclosure in API documentation.
+solution: >
+  Remove hardcoded secrets from API documentation, restrict access to API documentation endpoints,
+  and upgrade Swagger UI to a secure version. Ensure proper authentication is required to access documentation.
+references:
+  - https://swagger.io/docs/
+  - https://owasp.org/www-project-api-security/
+category: info_gather
+risk: high
+confidence: medium
+cweId: 200  # CWE-200: Exposure of Sensitive Information to an Unauthorized Actor
```

Comment:
CWE 200 isn't supposed to be mapped.
https://cwe.mitre.org/data/definitions/200.html
<img width="989" height="108" alt="image" src="https://github.com/user-attachments/assets/29a2ca49-f09c-445f-9878-d81cea493158" />

The following might be better options:
- https://cwe.mitre.org/data/definitions/497.html >  Exposure of Sensitive System Information to an Unauthorized Control Sphere 
- https://cwe.mitre.org/data/definitions/538.html > Insertion of Sensitive Information into Externally-Accessible File or Directory
- https://cwe.mitre.org/data/definitions/522.html > Insufficiently Protected Credentials